src/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -669,13 +669,70 @@ private object WritePrimitive(TypeMapping mapping, Func<string> readFunc)
                 string methodName = $"To{mapping.TypeDesc.FormatterName}";
                 if (!mapping.TypeDesc.HasCustomFormatter)
                 {
-                    MethodInfo method = typeof(XmlConvert).GetMethod(methodName, BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, new Type[] { typeof(string) });
```

Comment:
Just to confirm since the original reflection has `BindingFlags.NonPublic`, any chance the old code invokes non-public "To" function?